src/Adapter/Manufacturer/CommandHandler/AbstractManufacturerCommandHandler.php

```diff
@@ -0,0 +1,103 @@
+<?php
+/**
+ * 2007-2019 PrestaShop and Contributors
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2019 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShop\PrestaShop\Adapter\Manufacturer\CommandHandler;
+
+use Manufacturer;
+use PrestaShop\PrestaShop\Adapter\Manufacturer\AbstractManufacturerHandler;
+use PrestaShop\PrestaShop\Core\Domain\Manufacturer\Exception\DeleteManufacturerException;
+use PrestaShop\PrestaShop\Core\Domain\Manufacturer\Exception\ManufacturerException;
+use PrestaShop\PrestaShop\Core\Domain\Manufacturer\Exception\ManufacturerNotFoundException;
+use PrestaShop\PrestaShop\Core\Domain\Manufacturer\Exception\UpdateManufacturerException;
+use PrestaShop\PrestaShop\Core\Domain\Manufacturer\ValueObject\ManufacturerId;
+use PrestaShopException;
+
+/**
+ * Provides reusable methods for manufacturer command handlers
+ */
+abstract class AbstractManufacturerCommandHandler extends AbstractManufacturerHandler
+{
+    /**
+     * Deletes manufacturer
+     *
+     * @param ManufacturerId $manufacturerId
+     *
+     * @throws DeleteManufacturerException
+     * @throws ManufacturerException
+     * @throws ManufacturerNotFoundException
+     */
```

Comment:
@matks wdyt about these abstractions? The only thing currently separating single action from bulk is foreach loop. Could it cause any trouble?

Reply:
Looks good to me 👍 

One issue that could prevent the usage of such functions would be how to handle half-failure for bulk actions. If you ask to delete 10 manufacturers, the code is able to delete the 3 first manufacturers, but fails for the 4th (lets say because of a temporary database connection timeout), should the code keep deleting manufacturers and stop there ?

So far we've gone with the 2nd option - stop now as it might be risky to keep processing the command, so it works. If one day we modify this behavior, this will create issues. But we'll address this need when the time comes 😄